File: api/itf-scraper-cron.py
# Vercel Cron Job for ITF Scraper (Python)
# Runs 2x per day at 08:00 and 20:00 EET (06:00 and 18:00 UTC)
# Vercel Cron will call this endpoint automatically

import os
import sys
import asyncio
import logging
from pathlib import Path

# Add project root to path
project_root = Path(__file__).parent.parent
sys.path.insert(0, str(project_root))

from src.pipelines.itf_notion_pipeline import ITFNotionPipeline

logging.basicConfig(
    level=logging.INFO,
    format='%(asctime)s - %(levelname)s - %(message)s'
)

async def handler(req):
    """Vercel serverless function handler"""
    # Verify this is called by Vercel Cron (optional security)
    auth_header = req.headers.get('authorization', '')
    cron_secret = os.getenv('CRON_SECRET') or os.getenv('VERCEL_CRON_SECRET')
    
    if cron_secret and auth_header != f'Bearer {cron_secret}':
        return {
            'statusCode': 401,
            'body': {'error': 'Unauthorized'}
        }
    
    logging.info('🚀 ITF Scraper Cron started')
    
    try:
        # Run ITF pipeline
        config = {
            'scraper': {
                'target_tournaments': ['W15', 'W35', 'W50'],
                'rate_limit': 2.5,
            },
            'notion': {
                'tennis_prematch_db_id': None,  # Will load from env
            }
        }
        
        pipeline = ITFNotionPipeline(config)
        result = await pipeline.run_pipeline()
        
        if result.get('success'):
            logging.info(f"✅ Pipeline completed: {result['matches_created']} matches created")
            return {
                'statusCode': 200,
                'body': {
                    'success': True,
                    'timestamp': result.get('timestamp'),
                    'matches_scraped': result.get('matches_scraped', 0),
                    'matches_created': result.get('matches_created', 0),
                    'matches_duplicates': result.get('matches_duplicates', 0),
                    'matches_errors': result.get('matches_errors', 0),
                }
            }
        else:
            logging.error(f"❌ Pipeline failed: {result.get('error')}")
            return {
                'statusCode': 500,
                'body': {
                    'success': False,
                    'error': result.get('error'),
                    'timestamp': result.get('timestamp'),
                }
            }
    except Exception as e:
        logging.error(f'❌ ITF Scraper Cron Error: {e}', exc_info=True)
        return {
            'statusCode': 500,
            'body': {
                'success': False,
                'error': str(e),
            }
        }

# Vercel Python runtime expects this
def main(req):
    """Main entry point for Vercel"""
    return asyncio.run(handler(req))

File: api/itf-scraper-cron.js
// Vercel Cron Job for ITF Scraper
// Runs 2x per day at 08:00 and 20:00 EET (06:00 and 18:00 UTC)
// Vercel Cron will call this endpoint automatically
// This endpoint calls an external service or uses HTTP to trigger the Python script

const https = require('https');
const http = require('http');

export default async function handler(req, res) {
  // Verify this is called by Vercel Cron (optional security)
  const authHeader = req.headers.authorization;
  const cronSecret = process.env.CRON_SECRET || process.env.VERCEL_CRON_SECRET;
  
  if (cronSecret && authHeader !== `Bearer ${cronSecret}`) {
    return res.status(401).json({ error: 'Unauthorized' });
  }

  console.log('🚀 ITF Scraper Cron started at', new Date().toISOString());

  try {
    // Option 1: Call external service (if you have one running)
    // Option 2: Use Vercel's serverless function to call Python
    // For now, we'll use a simple HTTP call to trigger the scraper
    
    // Since Vercel doesn't support Python directly, we need to:
    // 1. Use an external service (Render, Railway, etc.) to run Python
    // 2. OR use Vercel's build command to install Python (not recommended)
    // 3. OR rewrite the scraper logic in Node.js
    
    // For Vercel Pro, the best approach is to:
    // - Keep the Python script running on a separate service
    // - Call it via HTTP from this cron endpoint
    
    const scraperUrl = process.env.ITF_SCRAPER_URL || 'https://your-python-service.com/run-scraper';
    
    // Make HTTP request to Python service
    const response = await fetch(scraperUrl, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        'Authorization': `Bearer ${process.env.SCRAPER_API_KEY || ''}`
      },
      body: JSON.stringify({
        action: 'run_itf_scraper',
        timestamp: new Date().toISOString()
      })
    });

    const result = await response.json();

    console.log('✅ ITF Scraper triggered successfully');

    return res.status(200).json({
      success: true,
      timestamp: new Date().toISOString(),
      message: 'ITF scraper triggered successfully',
      result: result
    });
  } catch (error) {
    console.error('❌ ITF Scraper Cron Error:', error);
    return res.status(500).json({
      success: false,
      error: error.message,
      timestamp: new Date().toISOString(),
      note: 'Make sure ITF_SCRAPER_URL is set in Vercel environment variables'
    });
  }
}
